group_vars/nextcloud_servers

```diff
@@ -220,6 +220,51 @@ devture_postgres_managed_databases_auto: |
 #                                                                      #
 ########################################################################
 
+######################################################################
+#
+# backup-borg
+#
+######################################################################
+
+backup_borg_enabled: false
+
+backup_borg_identifier: nextcloud-backup-borg
+
+backup_borg_base_path: "{{ nextcloud_base_path }}/backup_borg"
+
+backup_borg_username: "{{ nextcloud_user_username }}"
+backup_borg_uid: "{{ nextcloud_user_uid }}"
+backup_borg_gid: "{{ nextcloud_user_gid }}"
+
+backup_borg_retention_prefix: nextcloud-
+backup_borg_storage_archive_name_format: nextcloud-{now:%Y-%m-%d-%H%M%S}
+
+backup_borg_container_network: "{{ nextcloud_container_network }}"
+
+backup_borg_container_image_self_build: "{{ nextcloud_architecture not in ['amd64', 'arm32', 'arm64'] }}"
+
+backup_borg_postgresql_enabled: "{{ devture_postgres_enabled }}"
+backup_borg_postgresql_databases_hostname: "{{ devture_postgres_connection_hostname if devture_postgres_enabled else '' }}"
+backup_borg_postgresql_databases_username: "{{ devture_postgres_connection_username if devture_postgres_enabled else '' }}"
+backup_borg_postgresql_databases_password: "{{ devture_postgres_connection_password if devture_postgres_enabled else '' }}"
+backup_borg_postgresql_databases_port: "{{ devture_postgres_connection_port if devture_postgres_enabled else 5432 }}"
+backup_borg_postgresql_databases: "{{ devture_postgres_managed_databases | map(attribute='name') if devture_postgres_enabled else [] }}"
+
+backup_borg_location_source_directories:
+  - "{{ nextcloud_nextcloud_data_path }}"
```

Comment:
`nextcloud_nextcloud_data_path` is defined in `roles/custom/nextcloud_server/defaults/main.yml` like this:

```yaml
nextcloud_nextcloud_base_path: "{{ nextcloud_base_path }}/nextcloud"
nextcloud_nextcloud_data_path: "{{ nextcloud_nextcloud_base_path }}/data"
```

So by default we configure backups explicitly only for `/nextcloud/nextcloud/data`. While this is probably OK for backing up Nextcloud's stuff , users who enable other services may find that something is lacking.

We should probably back up all of `/nextcloud` (`nextcloud_base_path`) with the exception of `/nextcloud/postgres/data` or something.

The Matrix playbook skips over Postgres data with the following `backup_borg_location_exclude_patterns` exclusion: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/0b9dc56edffb3f85deb2919a7d4e2f9f69c0fe78/group_vars/matrix_servers#L1909-L1914

Reply:
Yes, that generally more sensible. Do you have a solution for users that have a `nextcloud_nextcloud_data_path` that is not part of `nextcloud_nextcloud_base_path`?
I think something like manually extending `backup_borg_location_source_directories` but can't get that working

Reply:
If the data path is outside the base path, then this is a very custom setup which breaks the assumptions that the playbook makes. Other stuff could break as well.

Feel free to adjust `backup_borg_location_source_directories` manually in that case.